### ISSUE TITLE
`default_config.nu` checkout file support not work

### ISSUE THREAD

**Issue by hustcer** [nushell/nushell#4599]
### Describe the bug

The following config in `default_config.nu` works for checkout branch but not work for checkout files:

```nu
extern "git checkout" [
  branch?: string@"nu-complete git branches" # name of the branch to checkout
  -b: string                                 # create and checkout a new branch
  ...
]
```

### How to reproduce

1. Run `git checkout main docs crates` will get:
```
$> git checkout main docs crates
Error: nu::parser::extra_positional (link)

  × Extra positional argument.
   ╭─[entry #14:1:1]
 1 │ git checkout main docs crates
   ·                   ──┬─
   ·                     ╰── extra positional argument
   ╰────
  help: Usage: git checkout {flags} (branch)

```
2. After adding `...files?: string                          # the file(s) to checkout` to the signature and run again:

![Screen Shot 2022-02-22 at 14 45 05](https://user-images.githubusercontent.com/1161607/155077379-d04733a3-625c-4d60-abfd-dae93c59fc4c.png)

### Expected behavior

Should work for checkout multiple files

### Configuration

| key                | value                                               |
| ------------------ | --------------------------------------------------- |
| version            | 0.59.0                                              |
| branch             | main                                                |
| short_commit       | 9888f8f2                                            |
| commit_hash        | 9888f8f298e18ca60e713637914014d8767e077c            |
| commit_date        | 2022-02-21 22:22:21 +00:00                          |
| build_os           | macos-x86_64                                        |
| rust_version       | rustc 1.58.1 (db9d1b20b 2022-01-20)                 |
| rust_channel       | stable-x86_64-apple-darwin                          |
| cargo_version      | cargo 1.58.0 (f01b232bc 2022-01-19)                 |
| pkg_version        | 0.59.0                                              |
| build_time         | 2022-02-22 09:51:58 +08:00                          |
| build_rust_channel | release                                             |
| features           | dataframe, default, trash, which, zip               |
| installed_plugins  | gstat, inc, query, query json, query web, query xml |

**Comment by kubouch** [comment]
Does the error appear even without `z nushell;`?

**Comment by hustcer** [comment]
> Does the error appear even without `z nushell;`?

Yes, of course

**Comment by jntrnr** [comment]
Yeah, we need to update the completion logic to also look for files and not just branches
